### PR TITLE
Aip 7108 graph ql survey create functionaliteiten unittests

### PR DESCRIPTION
**Pull request by anaim**

### CODE REVIEW

**Review by svbeusekom — APPROVED** [review]
Looks good, but tests are failing

File: src/schema/survey/survey.factory.spec.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import { Survey } from './models/survey.model';
+
+describe('SurveyFactory', () => {
+	test('CreateSurvey() constructs an instance of an Survey GraphQL model', () => {
+		const result = SurveyFactory.CreateSurvey(domainSurvey);
+		expect(result).toBeInstanceOf(Survey);
+		expect(result).toEqual(expect.objectContaining(survey));
+	});
+
+	test('CreateElementInput', () => {
```
Comment (fortunato) [inline src/schema/survey/survey.factory.spec.ts:12]: Can you fix the test name/description?

File: src/schema/survey/survey.resolver.spec.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+});
+
+const prismaServiceMock: MockedObjectDeep<PrismaService> = {
+	...(<any>{}),
+};
+
+const unitRepo = new UnitRepository(prismaServiceMock, new ManifestationRepository(prismaServiceMock));
+const elementRepo = new ElementRepository(prismaServiceMock, unitRepo);
+
+describe('Decomposition / Element / Resolver', () => {
```
[the remark this answers is not in the document]
Comment (anaim) [reply src/schema/survey/survey.resolver.spec.ts]: PR is updated! changes in van gekopieerd files waren niet opgeslagen!!

File: src/schema/survey/survey.factory.spec.ts (hunk 1 above)
Comment (anaim) [reply src/schema/survey/survey.factory.spec.ts]: PR is updated! changes in van gekopieerd files waren niet opgeslagen!!

File: src/schema/survey/survey.repository.spec.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+const repo = new SurveyRepository(prismaServiceMock);
+
+describe('SurveyRepository', () => {
+	test('createSurvey()', async () => {
+		await repo.createSurvey(surveyInput);
+		expect(prismaServiceMock.surveys.create).toHaveBeenCalledWith({
+			data: expect.objectContaining({
+				condition: '__CONDITION__',
+				description: '__DESCRIPTION__',
+				inspectionStandardType: '__INSPECTIONSTANDARDTYPE__',
```
Comment (fortunato) [inline src/schema/survey/survey.repository.spec.ts:29]: I think we should copy the `InspectionStandardTypes` enum from AIP for this

File: src/schema/survey/survey.repository.spec.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+			}),
+		});
+	});
+
+	test('getSurvey()', async () => {
+		const surveys = await surveyRepoMock.getSurveyById('0deb07f3-28f5-47e1-b72a-d1b2a19d4670');
+		expect(surveys).toEqual({
+			description: '__DESCRIPTION__',
+			id: '0deb07f3-28f5-47e1-b72a-d1b2a19d4670',
+			inspectionStandardType: '__INSPECTIONSTANDARDTYPE__',
```
Comment (fortunato) [inline src/schema/survey/survey.repository.spec.ts:47]: Can we make this an enum value?

File: src/schema/survey/survey.resolver.spec.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+		}
+	}),
+	...(<any>{}),
+});
+
+const prismaServiceMock: MockedObjectDeep<PrismaService> = {
+	...(<any>{}),
+};
+
+//const surveyRepo = new SurveyRepository(prismaServiceMock);
```
Comment (fortunato) [inline src/schema/survey/survey.resolver.spec.ts:39]: Can this be removed?

File: src/schema/survey/__stubs__/survey.ts [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+import { Survey } from '../models/survey.model';
+import { CreateSurveyInput } from '../dto/create-survey.input';
+import { DbSurvey as DomainSurvey } from '../types/survey.repository.interface';
+import { SurveyFactory } from '../survey.factory';
+
+const survey1 = new Survey();
+survey1.id = '0deb07f3-28f5-47e1-b72a-d1b2a19d4670';
+survey1.description = '__DESCRIPTION__';
+survey1.inspectionStandardType = '__INSPECTIONSTANDARDTYPE__';
```
Comment (fortunato) [inline src/schema/survey/__stubs__/survey.ts:11]: Can we use an enum for this for correctness?

File: src/schema/survey/__stubs__/survey.ts [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+survey2.id = 'ab8adf9e-65a0-40aa-9b1c-ea3537d1f8d7';
+survey2.description = 'BRU0002';
+
+export { survey1, survey2 };
+
+export const surveyRaw: Omit<DomainSurvey, 'id'> = {
+	objectId: 'f45c302c-6b18-85f6-bbe4-b3bf0a82d49a',
+	description: '__DESCRIPTION__',
+	condition: '__CONDITION__',
+	inspectionStandardType: '__INSPECTIONSTANDARDTYPE__',
```
Comment (fortunato) [inline src/schema/survey/__stubs__/survey.ts:24]: Can we use an enum for this for correctness?

File: src/schema/survey/__stubs__/survey.ts [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+export const surveyInput = Object.keys(surveyRaw).reduce((input, key) => {
+	input[key] = surveyRaw[key];
+	return input;
+}, new CreateSurveyInput());
+
+export const domainSurvey: DomainSurvey = {
+	...surveyRaw,
+	id: '0deb07f3-28f5-47e1-b72a-d1b2a19d4670',
+	description: '__DESCRIPTION__',
+	inspectionStandardType: '__INSPECTIONSTANDARDTYPE__',
```
Comment (fortunato) [inline src/schema/survey/__stubs__/survey.ts:91]: Can we use an enum for this for correctness?

File: src/schema/survey/survey.factory.spec.ts [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+		expect(result).toBeInstanceOf(Survey);
+		expect(result).toEqual(expect.objectContaining(survey));
+	});
+
+	test('CreateSurveyInput', () => {
+		const result = SurveyFactory.CreateSurvey(domainSurvey);
+		expect(result).toEqual({
+			description: '__DESCRIPTION__',
+			id: '0deb07f3-28f5-47e1-b72a-d1b2a19d4670',
+			inspectionStandardType: '__INSPECTIONSTANDARDTYPE__',
```
Comment (fortunato) [inline src/schema/survey/survey.factory.spec.ts:17]: Can we use an enum for this for correctness?

**Review by fortunato — CHANGES_REQUESTED** [review]

File: src/schema/survey/__stubs__/survey.ts (hunk 7 above)
Comment (anaim) [reply src/schema/survey/__stubs__/survey.ts]: Done

File: src/schema/survey/__stubs__/survey.ts (hunk 6 above)
Comment (anaim) [reply src/schema/survey/__stubs__/survey.ts]: Done

File: src/schema/survey/survey.resolver.spec.ts (hunk 5 above)
Comment (anaim) [reply src/schema/survey/survey.resolver.spec.ts]: Done

File: src/schema/survey/survey.repository.spec.ts (hunk 4 above)
Comment (anaim) [reply src/schema/survey/survey.repository.spec.ts]: Done

File: src/schema/survey/survey.repository.spec.ts (hunk 3 above)
Comment (anaim) [reply src/schema/survey/survey.repository.spec.ts]: Done

File: src/schema/survey/__stubs__/survey.ts (hunk 8 above)
Comment (anaim) [reply src/schema/survey/__stubs__/survey.ts]: Done

File: src/schema/survey/survey.factory.spec.ts (hunk 9 above)
Comment (anaim) [reply src/schema/survey/survey.factory.spec.ts]: Done

**Review by fortunato — APPROVED** [review]

**Pull request merged**